### PR TITLE
Mk 3033 email template mutation

### PR DESCRIPTION
**Pull request by rwhite27**
## General Description

Add email template and email template variables creation via grapql. This feature is expected for the admin.

## Related Issue

[Add Mutation for Email Template Modification](https://ottoradio.atlassian.net/browse/MK-3033)

## Checklist

A checklist of things that should be done before merging this PR. Linked to the ticket or issue that this PR is related to.(if applicable)

- [x] I have created a mutation for template creation
- [x] I added template variables to a template
- [x] I have created a migration that adds template_id to the email_templates_variables table.

### CODE REVIEW

File: graphql/schemas/Ecosystem/templates.graphql [hunk 1]
```diff
@@ -0,0 +1,59 @@
+type Templates {
+    id: ID!
+    users_id: Int!
```
Comment (kaioken) [inline graphql/schemas/Ecosystem/templates.graphql:3]: @rwhite27 no need for the id , use the user object and company object

File: graphql/schemas/Ecosystem/templates.graphql [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    companies_id: Int!
+    apps_id: Int!
```
Comment (kaioken) [inline graphql/schemas/Ecosystem/templates.graphql:5]: @rwhite27 no need for app

File: graphql/schemas/Ecosystem/templates.graphql [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    "companies_id"
+                    "apps_id"
+                    "name"
+                ]
+            )
+        orderBy: _ @orderBy(columns: ["id"], direction: "DESC")
+    ): [Templates!]!
+        @paginate(
+            model: "Kanvas\\Templates\\Models\\Templates"
+            scopes: ["fromCompany", "fromApp"]
```
Comment (kaioken) [inline graphql/schemas/Ecosystem/templates.graphql:57]: @rwhite27 add notDeleted scope

File: graphql/schemas/Ecosystem/templates.graphql [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+
+input TemplateVariableInput {
+    key: String!
+    value: String!
+}
+
+extend type Mutation @guard {
+    createOrUpdateTemplate(input: TemplatesInput!): Templates!
```
Comment (kaioken) [inline graphql/schemas/Ecosystem/templates.graphql:35]: @rwhite27 devide create and update even if we use the same method just for verbose

**Review by kaioken — CHANGES_REQUESTED** [review]

File: graphql/schemas/Ecosystem/templates.graphql (hunk 2 above)
Comment (rwhite27) [reply graphql/schemas/Ecosystem/templates.graphql:5]: why not?

**Review by FredPeal — APPROVED** [review]

**Pull request merged**